scripts/pylint_custom_plugin/pylint_guidelines_checker.py

```diff
@@ -1705,6 +1705,131 @@ def visit_functiondef(self, node):
+        ),
+        (
+            "ignore-enum-must-inherit-case-insensitive-enum-meta",
+            {
+                "default": False,
+                "type": "yn",
+                "metavar": "<y_or_n>",
+                "help": "Allow an enum to not inherit CaseInsensitiveEnumMeta.",
+            },
+        ),
+    )
+
+    def __init__(self, linter=None):
+        super(CheckEnum, self).__init__(linter)
+
+    def visit_classdef(self, node):
+        """Visits every enum class.
+
+        :param node: ast.ClassDef
+        :return: None
+        """
+        try:
+
+            inherits_metaclass, enum_class = self._check_is_enum_class(node)
+            
+            # If it has a metaclass, and is an enum class, check the capitalization
+            if inherits_metaclass and enum_class:
+                self._enum_uppercase(node)   
+            # Else if it does not have a metaclass, but it is an enum class
+            # Check both capitalization and throw pylint error for metaclass
+            elif enum_class:
+                self._does_not_inherit_case_insensitive(node)
+                self._enum_uppercase(node)   
+
+        except Exception:
+            logger.debug("Pylint custom checker failed to check enum.")
+            pass
+
+    def _check_is_enum_class(self,node):
+        """Checks if a class is an enum class.
+        Returns booleans for the two enum guidelines.
+
+        :param node: ast.ClassDef
+        :return: inherits_metaclass, enum_class
+        :rtype: bool, bool
+            
+        """
+
+        inherits_metaclass = False
+        enum_class = False
+
+        if node._metaclass:
+            if node._metaclass.name == "CaseInsensitiveEnumMeta":
+                inherits_metclass = True
+                enum_class = True
+                return inherits_metaclass, enum_class
+        for base in node.bases:
+            if isinstance(base, astroid.Call) and base.args[0].name == "CaseInsensitiveEnumMeta":
+                inherits_metclass = True
+                enum_class = True
+                return inherits_metaclass, enum_class
+            elif base.name == "Enum":
+                enum_class = True
+                return inherits_metaclass, enum_class
+                    
+                    
+        return False, False
+    
+    def _enum_uppercase(self, node):
+        """Visits every enum within the class.
+        Checks if the enum is uppercase, if it isn't it
+        adds a pylint error message.
+
+        :param node: ast.ClassDef
+        :return: None
+        """
+
+        # Check capitalization of enums assigned in the class
+        for nod in node.body:
+            if isinstance(nod, astroid.Assign):
+                if not nod.targets[0].name.isupper():
+                    self.add_message(
+                        "enum-must-be-uppercase", node=nod.targets[0], confidence=None
+                    )
+
+    def _does_not_inherit_case_insensitive(self, node):
```

Comment:
Since this function is only used in one place, it's best to remove it and put the contents where it is being called from.